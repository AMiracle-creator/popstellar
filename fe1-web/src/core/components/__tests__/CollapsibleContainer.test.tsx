import { describe } from '@jest/globals';
import { render } from '@testing-library/react-native';
import React from 'react';
import { Text } from 'react-native';

import CollapsibleContainer from '../CollapsibleContainer';

describe('CollapsibleContainer', () => {
  it('renders correctly when closed', () => {
    const component = render(
      <CollapsibleContainer title="A collapsible container">
        <Text>a first child</Text>
        <Text>a second child</Text>
      </CollapsibleContainer>,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });

  it('renders correctly when opened', () => {
    const component = render(
      <CollapsibleContainer title="A collapsible container" isInitiallyOpen>
        <Text>a first child</Text>
        <Text>a second child</Text>
      </CollapsibleContainer>,
    ).toJSON();
    expect(component).toMatchSnapshot();
  });
});
