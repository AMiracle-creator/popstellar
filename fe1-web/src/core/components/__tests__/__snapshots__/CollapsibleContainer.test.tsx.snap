// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollapsibleContainer container can be opened 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "black",
                "fontSize": 25,
                "fontWeight": "bold",
                "lineHeight": 26,
                "marginHorizontal": 10,
                "textAlign": "center",
              }
            }
          >
            open me
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 20,
                  "lineHeight": 26,
                  "marginHorizontal": 10,
                  "textAlign": "center",
                },
                Object {
                  "position": "absolute",
                  "right": 0,
                },
              ]
            }
          >
            v
          </Text>
        </View>
      </View>
    </RNGestureHandlerButton>
  </View>
  <Text>
    a first child
  </Text>
  <Text>
    a second child
  </Text>
</View>
`;

exports[`CollapsibleContainer renders correctly when closed 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "black",
                "fontSize": 25,
                "fontWeight": "bold",
                "lineHeight": 26,
                "marginHorizontal": 10,
                "textAlign": "center",
              }
            }
          >
            A collapsible container
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 20,
                  "lineHeight": 26,
                  "marginHorizontal": 10,
                  "textAlign": "center",
                },
                Object {
                  "position": "absolute",
                  "right": 0,
                },
              ]
            }
          >
            &lt;
          </Text>
        </View>
      </View>
    </RNGestureHandlerButton>
  </View>
</View>
`;

exports[`CollapsibleContainer renders correctly when opened 1`] = `
<View>
  <View
    accessible={true}
    collapsable={false}
    focusable={true}
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
  >
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
    >
      <View
        accessible={true}
        collapsable={false}
        nativeID="animatedComponent"
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View>
          <Text
            style={
              Object {
                "color": "black",
                "fontSize": 25,
                "fontWeight": "bold",
                "lineHeight": 26,
                "marginHorizontal": 10,
                "textAlign": "center",
              }
            }
          >
            A collapsible container
          </Text>
          <Text
            style={
              Array [
                Object {
                  "color": "#000",
                  "fontSize": 20,
                  "lineHeight": 26,
                  "marginHorizontal": 10,
                  "textAlign": "center",
                },
                Object {
                  "position": "absolute",
                  "right": 0,
                },
              ]
            }
          >
            v
          </Text>
        </View>
      </View>
    </RNGestureHandlerButton>
  </View>
  <Text>
    a first child
  </Text>
  <Text>
    a second child
  </Text>
</View>
`;
